### PR TITLE
Feat: Add Queue factory for uniqueness

### PR DESCRIPTION
**Pull request by steveworley**
This adds a new queue factory to give the Quant module some flexibility on how/when to add items to the queue. For now this extends the default queue handling to add a hash column for uniqueness. This will prevent duplicate queue items from being added when they already exist in the queue.

- Adds new factory and queue classes
- Updates queue calls to use the new factory
- Adds error message to the Drush command to notify of the new requirements

Note: This requires a change to `settings.php` for a site to utilise the Drush commands afterwards, the drush queue commands do not support different factories unless they're specified in settings.php.

You need to update `settings.php` with the following for the Drush commands to work:

```
$settings['queue_service_quant_seed_worker'] = 'quant.queue_factory';
```

### CODE REVIEW

**Comment by kepol** [comment]
Is this ready for review and testing?

**Comment by steveworley** [comment]
@kepol yep it should be good to go; no changes if you don't add the `settings.php` configuration change — we can't do that programmatically, so it has to be an advanced set up type deal.

**Review by stooit — APPROVED** [review]

**Comment by kepol** [comment]
Sorry I missed the comment in April but I see Stu has approved already so I assume there's nothing more to do here.

File: src/QuantQueueFactory.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+use Drupal\Core\Queue\QueueDatabaseFactory;
+use Drupal\Core\Site\Settings;
+
+/**
+ * {@inheritdoc}
+ */
+class QuantQueueFactory extends QueueDatabaseFactory {
+
+  /**
+   * Get an instance of the facotry.
```
Comment (kepol) [inline src/QuantQueueFactory.php]:
```suggestion
   * Get an instance of the factory.
```

File: src/QuantQueue.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  /**
+   * Length of the hash coulmn.
+   */
+  public const HASH_LENGTH = 48;
+
+  /**
+   * The algorithm used to hash the item.
+   */
+  public const HASH_METHOD = 'sha2';
```
Comment (kepol) [inline src/QuantQueue.php]: `sha2` used here and `sha512` elsewhere. Okay?

**Review by kepol — CHANGES_REQUESTED** [review]
I noticed a typo so took a quick look.

File: src/QuantQueue.php (hunk 2 above)
Comment (steveworley) [reply src/QuantQueue.php]: This actually isn't really used anywhere — you can't reverse the sha's this works by being the same as the previous entry. But let's make them consistent so there's no confusion here.

**Comment by kepol** [comment]
Released: https://github.com/quantcdn/drupal/releases/tag/1.5.2

**Pull request merged**